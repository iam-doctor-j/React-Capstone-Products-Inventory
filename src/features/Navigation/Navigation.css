/* .navbar {
    display: flex;
    flex-direction: row;
    height: 1.5rem;
    align-items: center;
}

.navbar ul {
    display: flex;
    flex-direction: row;
    list-style: none;
}

.navbar ul li {
    margin-right: 10px;
} */
.navbar-brand {
    letter-spacing: 0.75rem;
}