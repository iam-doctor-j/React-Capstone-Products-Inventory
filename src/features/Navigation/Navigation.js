import React, { useEffect } from 'react';
import { Link, NavLink, BrowserRouter as Router, Switch, Route, Redirect, Prompt } from 'react-router-dom';
import './Navigation.css';
import routes from '../../constants/routes';
import { connect } from 'react-redux';
import { logout, refresh } from '../Authentication/AuthenticationSlice';
import { ProductFeedLazy, ProductDetailsLazy, UserProfileLazy, RegisterLazy, LoginLazy, AddProductLazy, EditProductLazy, ChartLazy, AboutLazy } from '../../constants/LazyComponents';
import { ReactComponent as Loader } from '../../assets/suspense.svg';
import { toast } from 'react-toastify';

const PlaceHolder = () => {
    return(
        <div className="container h-100 d-flex justify-content-center align-items-center">
            <Loader />
        </div>
    )
}

export function Navigation(props) {
    const {loggedIn} = props;
    useEffect(() => {
        if(!loggedIn)
            props.dispatch(refresh());
    });

    return(
        <>
        <Router>
        <nav className="navbar navbar-light sticky-top bg-light navbar-expand-lg">
            <div className="container justify-content-start">
                <Link to={routes.HOME} className="navbar-brand mr-auto">Products Inventory</Link>
                <div className="ml-auto">
                    <ul className="navbar-nav">
                    { !loggedIn? 
                    <>
                        <li className="nav-item">
                            <NavLink className="nav-link" to={routes.LOGIN} activeClassName="active">
                                <span>Login <i className="fas fa-sign-in-alt"></i></span>
                            </NavLink>
                        </li>
                        <li className="nav-item">
                            <NavLink className="nav-link" to={routes.REGISTER} activeClassName="active">
                                <span>Register <i className="fas fa-user-plus"></i></span>
                            </NavLink>
                        </li>
                    </> : 
                    <>
                    <li className="nav-item">
                        <NavLink title="User Profile" className="nav-link" to={routes.USER} activeClassName="active">
                            <i className="fas fa-user"></i>
                        </NavLink>
                    </li>
                    <li className="nav-item">
                        <a title="Logout" className="nav-link" href="#" onClick={() => {props.dispatch(logout(() => toast.warn('Logged out successfully!')))}}>
                            <i className="fas fa-sign-out-alt"></i>
                        </a>
                    </li>
                    </>
                    }
                    <li className="nav-item">
                        <NavLink to={routes.ABOUT} title="About" className="nav-link" activeClassName="active">
                            <i className="fas fa-info-circle"></i>
                        </NavLink>
                    </li>
                    </ul>
                </div>
            </div>
        </nav>
        <React.Suspense fallback={<PlaceHolder />}>
        <Switch>
            <Route exact path={routes.HOME} component={ProductFeedLazy} />
            <Route path={routes.REGISTER} component={RegisterLazy} />
            {loggedIn!==null ? <Route path={routes.ADD_PRODUCTS} render={(props) => (loggedIn ?<AddProductLazy {...props}/> : <Redirect to="/login" />)}/> : <></>}
            {loggedIn!==null ? <Route path={routes.EDIT_PRODUCTS} render={(props) => (loggedIn ?<EditProductLazy {...props}/> : <Redirect to="/login" />)} /> : <></>}
            <Route path={routes.LOGIN} component={LoginLazy} />
            <Route path={routes.DETAILS} component={ProductDetailsLazy} />
            <Route path={routes.USER} component={UserProfileLazy} />
            <Route path={routes.CHART} component={ChartLazy} />
            <Route path={routes.ABOUT} component={AboutLazy}/>
        </Switch>
        </React.Suspense>
        </Router>
        </>
    );
}

export default connect(state => {
    return {
        loggedIn: state.auth.loggedIn,
        user: state.auth.user,
    }
})(Navigation);