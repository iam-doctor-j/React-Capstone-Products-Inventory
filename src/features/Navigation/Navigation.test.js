import React from 'react';
import { shallow } from 'enzyme';
import { Navigation } from './Navigation';
import { BrowserRouter } from 'react-router-dom';

describe('Testing Navigation with Shallow Rendering', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<Navigation/>);
    });

    it('Should render correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should contain one router', () => {
        expect(wrapper.find(BrowserRouter).length).toEqual(1);
    })

    it('should have a nav element', () => {
        expect(wrapper.find('nav').length).toEqual(1);
    })

    afterEach(() => {
        wrapper.unmount();
    });
})