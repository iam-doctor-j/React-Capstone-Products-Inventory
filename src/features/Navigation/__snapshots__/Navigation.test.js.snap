// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Navigation with Shallow Rendering Should render correctly 1`] = `
<Fragment>
  <BrowserRouter>
    <nav
      className="navbar navbar-light sticky-top bg-light navbar-expand-lg"
    >
      <div
        className="container justify-content-start"
      >
        <Link
          className="navbar-brand mr-auto"
          to="/"
        >
          Products Inventory
        </Link>
        <div
          className="ml-auto"
        >
          <ul
            className="navbar-nav"
          >
            <li
              className="nav-item"
            >
              <NavLink
                activeClassName="active"
                className="nav-link"
                to="/login"
              >
                <span>
                  Login 
                  <i
                    className="fas fa-sign-in-alt"
                  />
                </span>
              </NavLink>
            </li>
            <li
              className="nav-item"
            >
              <NavLink
                activeClassName="active"
                className="nav-link"
                to="/register"
              >
                <span>
                  Register 
                  <i
                    className="fas fa-user-plus"
                  />
                </span>
              </NavLink>
            </li>
            <li
              className="nav-item"
            >
              <NavLink
                activeClassName="active"
                className="nav-link"
                title="About"
                to="/about"
              >
                <i
                  className="fas fa-info-circle"
                />
              </NavLink>
            </li>
          </ul>
        </div>
      </div>
    </nav>
    <Suspense
      fallback={<PlaceHolder />}
    >
      <Switch>
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.lazy),
              "_ctor": [Function],
              "_result": null,
              "_status": -1,
            }
          }
          exact={true}
          path="/"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.lazy),
              "_ctor": [Function],
              "_result": null,
              "_status": -1,
            }
          }
          path="/register"
        />
        <Route
          path="/add"
          render={[Function]}
        />
        <Route
          path="/edit/:id"
          render={[Function]}
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.lazy),
              "_ctor": [Function],
              "_result": null,
              "_status": -1,
            }
          }
          path="/login"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.lazy),
              "_ctor": [Function],
              "_result": null,
              "_status": -1,
            }
          }
          path="/details/:id"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.lazy),
              "_ctor": [Function],
              "_result": null,
              "_status": -1,
            }
          }
          path="/user"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.lazy),
              "_ctor": [Function],
              "_result": null,
              "_status": -1,
            }
          }
          path="/chart"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.lazy),
              "_ctor": [Function],
              "_result": null,
              "_status": -1,
            }
          }
          path="/about"
        />
      </Switch>
    </Suspense>
  </BrowserRouter>
</Fragment>
`;
